repo2docker/contentproviders/zenodo.py

```diff
@@ -0,0 +1,99 @@
+import os
+import json
+import shutil
+
+from os import makedirs
+from os import path
+from urllib.request import urlopen, Request
+from zipfile import ZipFile, is_zipfile
+
+from .base import ContentProvider
+from ..utils import copytree
+
+
+class Zenodo(ContentProvider):
+    """Provide contents of a Zenodo deposit."""
+
+    def detect(self, doi, ref=None, extra_args=None):
+        """Trigger this provider for things that resolve to a Zenodo record"""
+        doi = doi.lower()
+        # 10.5281 is the Zenodo DOI prefix
+        if doi.startswith("10.5281/"):
+            resp = urlopen("https://doi.org/{}".format(doi))
+            self.record_id = resp.url.rsplit("/", maxsplit=1)[1]
+            return {"record": self.record_id}
+
+        elif doi.startswith("https://doi.org/10.5281/") or doi.startswith(
+            "http://doi.org/10.5281/"
+        ):
+            resp = urlopen(doi)
```

Comment:
Do we already use requests? If not maybe we should for this. I also think we should set a user-agent to be good citizens.

Reply:
We're already depending on requests because docker-py does.

Reply:
Ahhh! I had a look but didn't see `requests` in the direct dependencies and (re)learnt how `urlopen` works. Because I ended up using `urlopen` the tests now rely on being able to open a `file:///path/some/where/temporary/fake.zip` (for example [here](https://github.com/jupyter/repo2docker/pull/693/files#diff-6be528671d2ae05d5d68659f729a276bR64)) which we can't do with `requests`. How strong are your feelings on switching?

User-agent is a good idea, done.